src/matchRoute.js

```diff
@@ -38,10 +38,10 @@ function* matchRoute(route, baseUrl, path, parentParams) {
         params: match.params,
       };
 
-      for (const childRoute of route.children) {
+      for (let i = 0; i < route.children.length; i += 1) {
```

Comment:
`ForOfStatement` was added to `no-restricted-syntax` eslint rule https://github.com/airbnb/javascript/issues/1122